### PR TITLE
Make alertmanager config optional in nginx.conf 

### LINKED ISSUE #325: Possible not have alertmanager config hardcoded into nginx.conf
We want to set https://github.com/cortexproject/cortex-helm-chart/blob/master/values.yaml#L162 to false and set https://github.com/cortexproject/cortex-helm-chart/blob/master/values.yaml#L1060 to true.

But because of hard-coded valuess in https://github.com/cortexproject/cortex-helm-chart/blob/ad2f5cf3ce961af813fc73b9530be75a585322ac/templates/nginx/nginx-config.yaml#L88-L103 this doesn't seem possible.  

Is there some other way of doing this that anyone knows of?

### PR DESCRIPTION
**Pull request by WilliamCassanova**
**What this PR does**:
Optionally allows for alertmanager section of nginx.conf to not be included based upon the `alertermanager.enabled` flag.

**Which issue(s) this PR fixes**:
Fixes #325

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 * [ENHANCEMENT] Includes enable flags for each component #319
 
+## 1.3.1 / 2022-03-01
+
+* [ENHANCEMENT] Make altermanager config optional in nginx config #325
```
Comment (nschad) [inline CHANGELOG.md:9]: Don't declare a version yourself please. Just write your entry under master / unreleased

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,8 +43,8 @@
 * [BUGFIX] alertmanager/ruler deployment: fix indentation #266
 
 ## 1.0.0 / 2021-11-25
-### This Release includes BREAKING CHANGES
 
+### This Release includes BREAKING CHANGES
```
Comment (nschad) [inline CHANGELOG.md:47]: what happened here?

**Review by nschad — CHANGES_REQUESTED** [review]
Can you please also do this for the every distributor/alertmanager/ruler/.... endpoint? Otherwise this doesn't really make any sense

File: CHANGELOG.md (hunk 1 above)
Comment (WilliamCassanova) [reply CHANGELOG.md:6]: Done.

File: templates/nginx/nginx-config.yaml [hunk 3]
```diff
@@ -115,11 +125,15 @@ data:
           proxy_pass      http://{{ template "cortex.fullname" . }}-ruler.{{ $rootDomain }}$request_uri;
         }
 
+        {{- end }}
+
         # Config Config
         location ~ /api/prom/configs/.* {
           proxy_pass      http://{{ template "cortex.fullname" . }}-configs.{{ $rootDomain }}$request_uri;
         }
```
Comment (nschad) [inline templates/nginx/nginx-config.yaml:133]: Can we warp this as well even though it's deprecated. should be `configs.enabled`

**Review by nschad — CHANGES_REQUESTED** [review]

File: templates/nginx/nginx-config.yaml (hunk 3 above)
Comment (WilliamCassanova) [reply templates/nginx/nginx-config.yaml:135]: Done.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 ## master / unreleased
 
 * [ENHANCEMENT] Includes enable flags for each component #319
+* [ENHANCEMENT] Make altermanager,distributor,rules and config.config configurations optional in nginx config #325
```
Comment (nschad) [inline CHANGELOG.md:6]: I would rephrase it to "Exclude cortex components endpoint when disabled from nginx config"

Comment (nschad) [reply CHANGELOG.md]: Sorry @WilliamCassanova I mistyped. Should be 
> Exclude cortex components endpoint from nginx config when disabled

**Review by nschad — APPROVED** [review]

**Comment by WilliamCassanova** [comment]
So once this is approved do one of the folks in https://github.com/cortexproject/cortex-helm-chart/blob/master/MAINTAINERS.md have to merge it?  I would assume the answer it yes.

**Comment by nschad** [comment]
> So once this is approved do one of the folks in https://github.com/cortexproject/cortex-helm-chart/blob/master/MAINTAINERS.md have to merge it?  I would assume the answer it yes.

Waiting for 2nd approval and then we merge. Relax 😎

**Review by kd7lxl — APPROVED** [review]

**Comment by WilliamCassanova** [comment]
So when I create a Fork and it eventually gets merged back to `Master` as was done yesterday. what if anything causes the `Chart.yaml` version to update and what actually generated a new release and tag.

I think I read in one of the CONTRIBUTING.md docs that I should not update the version.

The reason I asked is I just attempted to test this today and I am noticing the same failure as before:
```
host not found in upstream "cortex-alertmanager.cortex.svc.cluster.local" in /etc/nginx/nginx.conf:72
nginx: [emerg] host not found in upstream "cortex-alertmanager.cortex.svc.cluster.local" in /etc/nginx/nginx.conf:72
```

**Comment by nschad** [comment]
> So when I create a Fork and it eventually gets merged back to `Master` as was done yesterday. what if anything causes the `Chart.yaml` version to update and what actually generates a new release and tag.
> 
> I think I read in one of the CONTRIBUTING.md docs that I should not update the version.
> 
> The reason I asked is I just attempted to test this today and I am noticing the same failure as before:
> 
> ```
> host not found in upstream "cortex-alertmanager.cortex.svc.cluster.local" in /etc/nginx/nginx.conf:72
> nginx: [emerg] host not found in upstream "cortex-alertmanager.cortex.svc.cluster.local" in /etc/nginx/nginx.conf:72
> ```

You need to wait for the release of a new version and deploy this helm chart from the master branch

**Comment by WilliamCassanova** [comment]
Thanks...I just wanted to make sure I hadn't missed something.  I can locally in the meantime so no problems.
